pytest-selenium/tests/test_attribution.py

```diff
@@ -128,9 +128,138 @@ def test_access_free_url_in_citation_text_shows_url_for_default_page(
     attribution.click_attribution_link()
 
     # THEN: The access for free at url in the the citation section should reference the default page of the book
-    default_page_slug = utility.get_default_page(book_slug)
+    default_page_slug = get_default_page(book_slug)
     attribution_access_free_url_expected = (
         "https://openstax.org/books/" + book_slug + "/pages/" + default_page_slug
     )
 
     assert attribution_access_free_url_expected == attribution.access_free_url
+
+
+@markers.test_case("C623700", "C623701")
+@markers.parametrize("page_slug", ["preface"])
+@markers.desktop_only
+@markers.nondestructive
+def test_license_details(selenium, base_url, page_slug):
+    """Verify license name, link address and copyright name"""
+    book_list = Library()
+    book_slugs = book_list.book_slugs_list
+
+    # Repeat the test for all books in the library
+    for book_slug in list(book_slugs):
+
+        # GIVEN: Book page is loaded
+        book = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+        attribution = book.attribution
+
+        # Skip any notification/nudge popups
+        while book.notification_present:
+            book.notification.got_it()
+
+        if book_slug in (
+            "business-law-i-essentials",
+            "calculus-volume-1",
+            "calculus-volume-2",
+            "calculus-volume-3",
+            "principles-financial-accounting",
+            "principles-managerial-accounting",
+        ):
+            license_name_expected = (
+                "Creative Commons Attribution-NonCommercial-ShareAlike License 4.0"
+            )
+            license_url_expected = "http://creativecommons.org/licenses/by-nc-sa/4.0/"
+
+        else:
+            license_name_expected = "Creative Commons Attribution License 4.0"
+            license_url_expected = "http://creativecommons.org/licenses/by/4.0/"
+
+        # WHEN: The attribution section is expanded
+        attribution.click_attribution_link()
+
+        # THEN: License name & license link address is Creative Commons
+        license_name = attribution.citation_builder.get_attribute("text").strip()
+        license_url = attribution.citation_builder.get_attribute("href").strip()
+
+        try:
+            assert (
+                license_name == license_name_expected
+            ), f"{book_slug} has license name as {license_name}. But expected name is {license_name_expected}"
```

Comment:
You're not likely to see this message because the `try` will immediately catch the error. Instead, remove the `try` block and move both boolean statements into a single `assert`.

Reply:
I want the statement to be printed only if `license_name == "cc by"`. The test can still pass but I just need the list of books with that name. So is this ok?

Reply:
In that case just remove the failure string from the first assertion as you won't see it - it won't print if it passes and it won't print if it fails because the assertion error will be caught and handled.